packages/flutter/test/material/dropdown_form_field_test.dart

```diff
@@ -195,6 +195,162 @@ void main() {
     expect(value, equals('three'));
   });
 
+  testWidgets('DropdownButtonFormFieldState value updates when a different option is selected', (WidgetTester tester) async {
+    const String value = 'one';
+    final GlobalKey<FormFieldState<String>> stateKey = GlobalKey<FormFieldState<String>>();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: DropdownButtonFormField<String>(
+            key: stateKey,
+            value: value,
+            onChanged: (String newValue) {},
+            items: menuItems.map((String value) {
+              return DropdownMenuItem<String>(
+                value: value,
+                child: Text(value),
+              );
+            }).toList(),
+          ),
+        ),
+      )
+    );
+
+    // [FormField] value is properly initialized.
+    expect(stateKey.currentState.value, equals('one'));
+
+    // Tap [DropdownButton] to open dropdown menu.
+    await tester.tap(find.text('one'));
+    await tester.pumpAndSettle();
+
+    // Tap [DropdownMenuItem] for an item other than the initial item.
+    await tester.tap(find.text('three').last);
+    await tester.pumpAndSettle();
+
+    // [FormField] should update to selected item value.
+    expect(stateKey.currentState.value, equals('three'));
+  });
+
+  testWidgets("DropdownButtonFormField's InputDecorator isEmpty behavior is correct", (WidgetTester tester) async {
+    final GlobalKey<FormFieldState<String>> stateKey = GlobalKey<FormFieldState<String>>();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: DropdownButtonFormField<String>(
+            key: stateKey,
+            value: null, // setting value to null sets isEmpty to true
+            decoration: const InputDecoration(labelText: 'label', alignLabelWithHint: true),
+            onChanged: (String newValue) {},
+            items: menuItems.map((String value) {
+              return DropdownMenuItem<String>(
+                value: value,
+                child: Text(value),
+              );
+            }).toList(),
+          ),
+        ),
+      )
+    );
+
+    // isEmpty: true causes the label to be aligned with the input text if alignLabelWithHint is set to true
+    await tester.pumpAndSettle();
+    expect(tester.getSize(find.byType(InputDecorator)), const Size(800.0, 600.0));
+    expect(tester.getTopLeft(find.text('label')).dy, 32.0);
+    expect(tester.getBottomLeft(find.text('label')).dy, 48.0);
+    expect(
+      tester.getTopLeft(find.byType(InputDecorator)).dy,
+      tester.getTopLeft(find.text('label')).dy - 32.0
+    ); // Label is lower by 32 pixels than the InputDecorator, in the center of the TextField
```

Comment:
Your expect statement does not match up with the code comment. The code comment makes it sound like it's supposed to be something like: 
```dart
expect(
      tester.getTopLeft(find.text('label')).dy,
      // (bottom of text field - top of text field) / 2.0
    ); // Label is lower by 32 pixels than the InputDecorator, in the center of the TextField
```

If that's the case, the code should also be updated later in the test to properly reflect this.